### ISSUE TITLE
After DoctrineProxyFilter it is not possible to apply further filters

### ISSUE THREAD

**Issue by cschroeerolb** [myclabs/DeepCopy#98]
As  @IMM0rtalis mentioned it in his comment https://github.com/myclabs/DeepCopy/issues/18#issuecomment-306771518 there is in my opinion an undesired behavior with following code:

```php
$copyHelper = new DeepCopy();
$copyHelper->addFilter(new DoctrineProxyFilter(), new DoctrineProxyMatcher());
$copyHelper->addFilter(new SetNullFilter(), new PropertyNameMatcher('id'));
```

The DeepCopy-Class loads the proxy, but do not apply further filters like to set null the id. Therefore, Doctrine can not handle the copied entity as a new entity.

Do I see it correclty?

Best,
Christoph

**Comment by theofidry** [comment]
Hm if the proxy is loaded then the actual value should be deep-copied and property filters applied to it so if that's not the case there's an issue IMO. Could you do a PR with a failing test case?

@mnapoli could you confirm this?

**Comment by mnapoli** [comment]
Sorry what you say makes sense but I'm not really sure :/ (it's been a while since I coded these classes)

**Comment by theofidry** [comment]
It's ok, as long as it makes sense to you at least I know it's somewhat the right direction :P

**Comment by theofidry** [comment]
It's ok, as long as it makes sense to you at least I know it's somewhat the right direction :P

**Comment by cschroeerolb** [comment]
I commited a (symfony-)project [DoctrineProxyFilterExample]( https://github.com/cschroeerolb/DoctrineProxyFilterExample ) with a Unit-Test DefaultServiceTest (with Proxy and without Proxy). For the test with the Proxy I used the ` getReference`-Method of the Entity Manager to work with a Proxy directly. The test with a Proxy fails because the ids are not null. 

I hope this helps and Thanks!

Best, 
Christoph
